cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -237,6 +240,22 @@ private void addInput(String inputName, InputFormatProvider inputFormatProvider,
     if (mapperClass != null && !Mapper.class.isAssignableFrom(mapperClass)) {
       throw new IllegalArgumentException("Specified mapper class must extend Mapper.");
     }
+    // Record lineage only if the input format has a proper name
+    if (!inputName.equals(inputFormatProvider.getInputFormatClassName())) {
+      try {
+        // Create an external dataset for the input format for lineage tracking
```

Comment:
You are right, I'll remove the suffix from dataset name. Having a suffix is not necessary if we use datgaset runtime args to pass configuration around.